index.bs

```diff
@@ -254,37 +322,46 @@ is determined by the platform and will be left to the platform.
+4. Let |pictureInPictureBrowsingContext| be a new <a>browsing context</a> with the
+    <a>opener browsing context</a> set to the current document's browsing context.
+5. Let |document| be a new {{Document}} with its browsing context set to
+    |pictureInPictureBrowsingContext| and the URL set to <a>about:blank</a>.
+5. Let |clone children flag| be `true`.
+6. Run <a>node clone</a> with |element|, |document| and |clone children flag|.
+7. Run the <a>update Picture-in-Picture element algorithm</a> with |element| and
+    |document|.
+
 # Video Specific Concepts # {#video-concepts}
 
-## Request Picture-in-Picture ## {#request-pip}
+## Request Video Picture-in-Picture ## {#request-video-pip}
 
-When the <dfn>request Picture-in-Picture algorithm</dfn> with |video|,
+When the <dfn>request video Picture-in-Picture algorithm</dfn> with |video|,
 |userActivationRequired| and |playingRequired| is invoked, the user agent MUST
 run the following steps:
 
-1. If <a>Picture-in-Picture support</a> is `false`, throw a
-    {{NotSupportedError}} and abort these steps.
-2. If the document is not allowed to use the <a>policy-controlled feature</a>
-    named `"picture-in-picture"`, throw a {{SecurityError}} and abort these
-    steps.
-3. If |video|'s {{readyState}} attribute is {{HAVE_NOTHING}}, throw a
+1. Run the <a>check Picture-in-Picture is allowed algorithm</a> with |video| and
+    |userActivationRequired|. If an error is thrown then throw this error.
+2. If |video|'s {{readyState}} attribute is {{HAVE_NOTHING}}, throw a
     {{InvalidStateError}} and abort these steps.
-4. If |video| has no video track, throw a {{InvalidStateError}} and abort
+3. If |video| has no video track, throw a {{InvalidStateError}} and abort
     these steps.
-5. OPTIONALLY, if the {{disablePictureInPicture}} attribute is present on
+4. OPTIONALLY, if the {{disablePictureInPicture}} attribute is present on
     |video|, throw an {{InvalidStateError}} and abort these steps.
-6. If |userActivationRequired| is `true` and the algorithm is not
-    <a>triggered by user activation</a>, throw a {{NotAllowedError}} and
-    abort these steps.
-7. If |video| is {{pictureInPictureElement}}, abort these steps.
-8. If |playingRequired| is `true` and |video| is {{paused}}, abort these steps.
-9. Set {{pictureInPictureElement}} to |video|.
-10. Let <dfn>Picture-in-Picture window</dfn> be a new instance of
-    {{PictureInPictureWindow}} associated with {{pictureInPictureElement}}.
-11. <a>Queue a task</a> to <a>fire an event</a> with the name
-    {{enterpictureinpicture}} at the |video| with its {{bubbles}} attribute
-    initialized to `true` and its {{pictureInPictureWindow}} attribute
-    initialized to <a>Picture-in-Picture window</a>.
+5. If |video| is {{pictureInPictureElement}}, abort these steps.
+6. If |playingRequired| is `true` and |video| is {{paused}}, abort these steps.
+7. Run the <a>update Picture-in-Picture element algorithm</a> with |video|.
```

Comment:
Do we intentionally miss to pass `|Document|`?

Reply:
Yes there is no document for video pip